src/Bicep.Cli/Services/TemplateEvaluation.cs

```diff
@@ -176,19 +193,42 @@ private static JToken EvaluateTemplate(JToken? templateJtoken, JToken? parameter
 
                 TemplateEngine.ValidateProcessedTemplate(template, "2020-10-01", deploymentScope);
 
-                return template.ToJToken();
+                assertions = template.Asserts?.ToDictionary(p => p.Key, p => (bool)p.Value.Value) ?? new Dictionary<string, bool>();
+
+                resultTemplate = template;
+
+                return true;
             }
             catch (Exception exception)
             {
-                throw new InvalidOperationException(
+                error = new InvalidOperationException(
                     $"Evaluating template failed: {exception.Message}." +
                     $"\nTemplate file: {templateJtoken}" +
                     (parametersJToken is null ? "" : $"\nParameters file: {parametersJToken}"),
                     exception);
+
+                return false;
+            }
+        }
+
+        private bool wasTestSuccessfull()
```

Comment:
nit: I'd avoid using past tense.
```suggestion
        private bool IsTestSuccessfull()
```